### DISCUSSION TITLE
Show point in circle when on streak

### DISCUSSION THREAD

**Discussion by sojusnik** [iSoron/uhabits#624] [Feature Requests]
While in the main view, a point in the circle would show if you're still on a streak and therefore further motivating you to stick to the habit.

Something like:
![grafik](https://user-images.githubusercontent.com/1245718/90955177-9f1fe780-e47b-11ea-9b0f-13f95eaf418f.png)

**Comment by iSoron** [comment]
Thank you for the suggestion, @sojusnik. What do you mean by "still on a streak"? This circle would show up if the current streak is the longest one so far?

**Comment by stellarbeam** [comment]
I think what he means by "still on streak" is if the streak is longer than 1 day. Is that correct @sojusnik ?

Maybe a smaller circle (than shown) in between could look visually better. A number in the filled circle could be yet more informational. Increasing the overall circle size might an option. Keeping the interface clutter-free should be given priority.
I love how clutter the experience already is..

**Comment by chennemann** [comment]
At first I thought this is a really cool idea but in the end it will not really provide any clear benefits in my opinion. You already know whether you are on a streak or not by looking at the checkmarks right next to the circle. And having a special indicator for the case you are currently on the longest streak seems a little bit overkill to me. 
So from my point of view this feature is not really feasible.

**Comment by sojusnik** [comment]
> What do you mean by "still on a streak"?

In the following screenshot you'll see a habit that has to be done twice a week. If you stay within the desaturated days, see yellow arrow, then you are "on a streak", analogous to the "don't break the chain" method.

![image](https://user-images.githubusercontent.com/1245718/91153520-9cfa9a80-e6c0-11ea-9d24-4e9cf67c0e6c.png)

> Maybe a smaller circle (than shown) in between could look visually better.

Sure, whatever looks best. That's just a draft :)

> A number in the filled circle could be yet more informational. Increasing the overall circle size might an option.

I'm against a number, because it would be hard to read and this information isn't that important to show it in the main view. The main idea here is to have an indicator for showing you that you are on track with your habits. This is one of the core motivational aspects of Jerry Seinfeld's "don't break the chain" method.

> You already know whether you are on a streak or not by looking at the checkmarks right next to the circle.

This is only true when you have to carry out a habit every day. But I do have a lot of habits which are due only 2 or 3 times a week, some even once a month. In this case my proposal would really help and is easy to implement.

**Comment by chennemann** [comment]
This:
> In the following screenshot you'll see a habit that has to be done twice a week. If you stay within the desaturated days, see yellow arrow, then you are "on a streak", analogous to the "don't break the chain" method.

and this:
> This is only true when you have to carry out a habit every day. But I do have a lot of habits which are due only 2 or 3 times a week, some even once a month. In this case my proposal would really help and is easy to implement.

are contradictory in my opinion. As long as we are in the desaturated days we know we are on a streak because we see the implicit checkmarks. Only when we (would) break the streak the cross is shown. So I stand by my point that the dot does not necessarily give any additional useful information that we could not infer anyways. 
However I do think it could look nice. So if you want you could just create a PR with that feature implemented (that should be relatively straight forward and unintrusive to the rest of the codebase). 
My stance is that we probably should not plan this as a feature but if someone wants to implement it anyways I don't see a problem with incorporating it. @iSoron what do you think?

**Comment by sojusnik** [comment]
> As long as we are in the desaturated days we know we are on a streak because we see the implicit checkmarks. Only when we (would) break the streak the cross is shown.

The thing is that those implicit checkmarks are only useful for non-daily habits. If you have a daily habit, then you'll still see a cross at the current day, before ticking this habit off and therefore remaining within the streak. Additionally, when using the dark mode, those implicit checkmarks and crosses are hard to recognize, while my proposed indicator should be clearly visible to motivate people not to break the chain/streak.

> So if you want you could just create a PR with that feature implemented (that should be relatively straight forward and unintrusive to the rest of the codebase).

Sorry, I don't know how to do this.
